chasm/tree.go

```diff
@@ -153,6 +164,210 @@ func NewEmptyTree(
 	return root
 }
 
+// setValue sets node value field and initialize serializedValue field with empty attributes based on fieldType.
+func (n *Node) setValue(v any, ft fieldType) error {
+	if err := validateValueType(v); err != nil {
+		return err
+	}
+
+	n.value = v
+
+	switch ft {
+	case fieldTypeData:
+		n.serializedValue = &persistencespb.ChasmNode{
+			Metadata: &persistencespb.ChasmNodeMetadata{
+				InitialVersionedTransition: &persistencespb.VersionedTransition{
+					TransitionCount:          n.backend.NextTransitionCount(),
+					NamespaceFailoverVersion: n.backend.GetCurrentVersion(),
+				},
+				Attributes: &persistencespb.ChasmNodeMetadata_DataAttributes{
+					DataAttributes: &persistencespb.ChasmDataAttributes{},
+				},
+			},
+		}
+	case fieldTypeComponent:
+		n.serializedValue = &persistencespb.ChasmNode{
+			Metadata: &persistencespb.ChasmNodeMetadata{
+				InitialVersionedTransition: &persistencespb.VersionedTransition{
+					TransitionCount:          n.backend.NextTransitionCount(),
+					NamespaceFailoverVersion: n.backend.GetCurrentVersion(),
+				},
+				Attributes: &persistencespb.ChasmNodeMetadata_ComponentAttributes{
+					ComponentAttributes: &persistencespb.ChasmComponentAttributes{},
+				},
+			},
+		}
+	case fieldTypeComponentPointer:
+		panic("not implemented")
+	}
+	return nil
+}
+
+func validateValueType(v any) error {
+	return validateType(reflect.TypeOf(v))
+}
+
+func validateType(t reflect.Type) error {
+	// TODO: for component, interface should also be supported.
+	if t.Kind() != reflect.Ptr && t.Elem().Kind() != reflect.Struct {
+		return serviceerror.NewInternal("only pointer to struct is supported for tree node value")
+	}
+	return nil
+}
+
+func fieldName(f reflect.StructField) string {
+	if tagName := f.Tag.Get(fieldNameTag); tagName != "" {
+		return tagName
+	}
+	return f.Name
+}
+
+func (n *Node) setSerializedValue(
```

Comment:
same comment as the `serializedValue` field. It's feels very confusing to me.

Reply:
renamed to `setSerializedNode()`.